tests/tests_pytorch/trainer/connectors/test_accelerator_connector.py

```diff
@@ -970,41 +1020,33 @@ def _mock_tpu_available(value):
         assert connector.strategy.launcher is None
 
     # Single HPU
-    if _lightning_habana_available():
-        import lightning_habana
-
-        with monkeypatch.context():
-            mock_cuda_count(monkeypatch, 0)
-            mock_mps_count(monkeypatch, 0)
-            mock_tpu_available(monkeypatch, False)
-            mock_ipu_available(monkeypatch, False)
-            mock_hpu_available(monkeypatch, True)
-            monkeypatch.setattr(lightning_habana.accelerator.HPUAccelerator, "auto_device_count", lambda *_: 1)
-            connector = _AcceleratorConnector()
-        assert isinstance(connector.accelerator, HPUAccelerator)
-        assert isinstance(connector.strategy, SingleHPUStrategy)
-        assert connector._devices_flag == 1
+    with monkeypatch.context():
+        mock_cuda_count(monkeypatch, 0)
+        mock_mps_count(monkeypatch, 0)
+        mock_tpu_available(monkeypatch, False)
+        mock_ipu_available(monkeypatch, False)
+        mock_hpu_count(monkeypatch, 1)
+        connector = _AcceleratorConnector()
+    assert isinstance(connector.accelerator, HPUAccelerator)
+    assert isinstance(connector.strategy, SingleHPUStrategy)
+    assert isinstance(connector.precision_plugin, MockHPUPrecisionPlugin)
+    assert connector._devices_flag == 1
 
     monkeypatch.undo()  # for some reason `.context()` is not working properly
     _mock_interactive()
 
-    if not is_interactive and _lightning_habana_available():  # HPU does not support interactive environments
-        from lightning_habana import HPUParallelStrategy
-
+    if not is_interactive:  # HPU does not support interactive environments
```

Comment:
Why did the assertions get removed then?

Reply:
Because mocking them is annoying and not valuable. But I can add them back and run them only if `_LIGHTNING_HABANA_AVAILABLE`

Reply:
Done